Exiled.Events/Patches/Events/Map/ExplodingFlashGrenade.cs

```diff
@@ -38,10 +38,13 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
         {
             List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
 
-            // Remove check if player is thrower. Grenade on self should affect themself.
-            int removeSelfCheckOffset = -4;
-            int removeSelfCheck = newInstructions.FindIndex(instruction => instruction.LoadsField(Field(typeof(Footprint), nameof(Footprint.Hub)))) + removeSelfCheckOffset;
-            newInstructions.RemoveRange(removeSelfCheck, 7);
+            if (Exiled.Events.Events.Instance.Config.CanFlashBangsAffectThrower)
```

Comment:
If you already changed `ProcessPlayers`, why need to change this patch? if `Config.CanFlashBangsAffectThrower` will be false, same player never been flashed,  `reload cfgs` with `Config.CanFlashBangsAffectThrower` equals to `true`, will not affect this

I think check in `ProcessPlayers` will be enough, and `if (Exiled.Events.Events.Instance.Config.CanFlashBangsAffectThrower)` in transpiler cannot support dynamic update of configs

Reply:
I would recommend you look at the location being patched

![image](https://user-images.githubusercontent.com/24619207/174443705-07ed7f16-e37f-418e-be3e-1927a6226dde.png)

If CanFlashBangsAffectThrower is true, patch will be applied (removal of NW self-check), and you can reach ProcessPlayer (for self-flash)
If CanFlashBangsAffectThrower is false, patch will not be applied, and you won't reach ProcessPlayer (for self-flash)

I am aware these configs are not hot-swappable, that's on the user to set it up as they desire. They can always add their own plugin to modify the function themselves. 

Furthermore, the check within ProcessPlayers is overkill, but I considered that in case someone called that function directly for some reason, the check should be there to ensure consistent behavior.

Reply:
> If CanFlashBangsAffectThrower is true, patch will be applied (removal of NW self-check), and you can reach ProcessPlayer (for self-flash)
> If CanFlashBangsAffectThrower is false, patch will not be applied, and you won't reach ProcessPlayer (for self-flash)

Yes, i said about this, I think we need exactly dynamic configurations

> I am aware these configs are not hot-swappable, that's on the user to set it up as they desire. They can always add their own plugin to modify the function themselves.

I don't see why we can't do it dynamically if we can? Besides, we can just save your changes in `ProcessPlayers` and it will work perfectly

Reply:
> hey can always add their own plugin to modify the function themselves.

I believe I understand your point, keep the removal of lines that the transpiler handle the logic. My only thought process would be for the ExplodingGrenadeEventArgs but I think you're right - it would be fine to keep the line removal and keep process player logic

Reply:
Yep, I'm convinced, it should work. Thanks for the feedback!

Reply:
> > hey can always add their own plugin to modify the function themselves.
> 
> I believe I understand your point, keep the removal of lines that the transpiler handle the logic. My only thought process would be for the ExplodingGrenadeEventArgs but I think you're right - it would be fine to keep the line removal and keep process player logic

Yea, that's what I mean

> Yep, I'm convinced, it should work. Thanks for the feedback!

My pleasure